tests/foreign_object/tests.py

```diff
@@ -767,3 +768,12 @@ def test_pickling_foreignobject(self):
         foreign_object_restored = pickle.loads(pickle.dumps(foreign_object))
         self.assertIn("path_infos", foreign_object_restored.__dict__)
         self.assertIn("reverse_path_infos", foreign_object_restored.__dict__)
+
+
+class ForeignObjectConstraintTests(TestCase):
+    @skipUnlessDBFeature("supports_table_check_constraints")
+    def test_validate_constraints_with_foreign_key_id(self):
+        patron_tab = PatronTab(patron_id=None)
+        with self.assertRaises(ValidationError) as cm:
+            patron_tab.validate_constraints()
```

Comment:
Could you add a test case showing the constraint being validated in the success case with an `assertNumQueries(1)`? This would catch the performance regression I mentioned.

Reply:
Just a note on the third comment: in practice the query count comes out as 3 rather than 1:

```sql
SAVEPOINT s281473600110624_x2
SELECT 1 AS _check WHERE COALESCE((500 < 1000), 1) = 1
RELEASE SAVEPOINT s281473600110624_x2
```

So even though there’s only a single SELECT, the additional savepoint statements bump the count. SQLite passes since it doesn’t add these, but on other databases we see this difference. Do you have any suggestions on how we should handle this in the test?